README.md

```diff
@@ -146,6 +147,141 @@ Both `textInputAreaView` and `textInputView` must be created in order for `NMess
+2. Pass the user's input to the networking service. This is achieved as a plain method call.
+
+Intercepting the user's input might be not quite obvious since you do not need any delegate subscriptions. It is done by overriding the `NMessengerViewController.sendText()` instance method.
+
+
+```swift
+public class MyChatMessagingVC: NMessengerViewController, IChatServiceDelegate
+{
+    override func sendText(_ text: String, isIncomingMessage:Bool) -> GeneralMessengerCell
+    {
+         let shouldSendToServer = !isIncomingMessage
+
+         if (shouldSendToServer)
+         {
+            // trigger network service
+            self.controller?.sendMessageAsync(text)
+         }
+
+         // otherwise - just render 
+         // `super` is critical to avoid recursion and "just render"
+         return super.sendText(text, isIncomingMessage: isIncomingMessage)
+    }
+}
+```
+
+I'd like to highlight the importance of using `super.sendText()` at the end of the function. If `self` is used in this case, you're going to
+1. end up with infinite recursion
+2. eventually crash due to "stack overflow" reason
+3. flood the chat with repeated messages
+
+You can use some other cell contruction code instead. See the ["Content Nodes and Custom Components"](https://github.com/eBay/NMessenger#content-nodes-and-custom-components) section for details.
+
+
+
+We can break down the process of "receiving a message" in two phases as well. Here they are: 
+1. Subscribe to events from your service. Usually it's done by one of iOS mechanics such as delegates, `NSNotification` or closures.
+2. Render the message using `NMessenger`
+
+Let's see how it can be done :
+
+```swift
+public class MyChatMessagingVC: NMessengerViewController, IChatServiceDelegate
+{
+      func chatService(_ sender: IChatService, didReceiveMessage message: IChatMessage)
+      {
+           // Using `super` to avoid side effects.
+           //
+           super.sendText(message.text, isIncomingMessage: true)
+      }
+}
+```
+
+So, now your app should be ready to process realtime messaging.
+
+
+
+When it comes to history handling, one of the approaches might be purging all the messages from `NMessenger` and re-adding them.
+
+```swift
+public class MyChatMessagingVC: NMessengerViewController, IChatServiceDelegate
+{
+      override func viewDidAppear(_ animated: Bool)
+      {
+            super.viewDidAppear(animated)
+
+            // supposing `self._chatService` has been the setup in `viewDidLoad` 
+            // or injected during the screen transition
+            //
+            self._chatService.loadHistoryAsync()
+      }
+
+
+      func chatService(_ sender: IChatService, didReceiveHistory: [IChatMessage]])
+      {
+            super.clearALLMessages()
+
+            messageList.forEach
+            {
+                // using `super`to avoid side effects
+                //
+                _ = super.sendText($0.text, isIncomingMessage: $0.isIncoming)
+            }
+      }
```

Comment:
@atainter , i am sorry but i am still unsure how to extend my PR properly. I would be grateful if you elaborated on adding the missing parts and posted the result to a readme (or a wiki page or some other documentation piece for this library).